CI/BUG?: test_pct_max_many_rows crashes on travis-27.

Following Ian Fleming ("Once is happenstance. Twice is coincidence. Three times is enemy action"), here are the three most recent builds from #23582 -- which *only* changes tests -- wherein there is a single failure/crash in the `travis-27.yaml` build for `pandas/tests/series/test_rank.py::test_pct_max_many_rows`, which looks like this:

```
[...]
........................................................................ [ 87%]
..........................................................[gw0] node down: Not properly terminated
fReplacing crashed worker gw0
gw2 [39907] / gw1 [39907]. [ 87%]
........................................................................ [ 87%]
........................................................................ [ 87%]
[...]
........................................................................ [ 99%]
.....
=================================== FAILURES ===================================
_______________________ pandas/tests/series/test_rank.py _______________________
[gw0] linux2 -- Python 2.7.15 /home/travis/miniconda3/envs/pandas/bin/python
Worker 'gw0' crashed while running 'pandas/tests/series/test_rank.py::test_pct_max_many_rows'
```

https://travis-ci.org/pandas-dev/pandas/jobs/455175414
https://travis-ci.org/pandas-dev/pandas/jobs/455259318
https://travis-ci.org/pandas-dev/pandas/jobs/455642129

Interestingly, the azure 2.7 builds pass both for windows and linux. Any ideas?

--- a/pandas/tests/frame/test_rank.py
+++ b/pandas/tests/frame/test_rank.py
@@ -310,6 +310,7 @@ class TestRank(TestData):
         expected = DataFrame(exp)
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.single
     def test_pct_max_many_rows(self):
         # GH 18271
         df = DataFrame({'A': np.arange(2**24 + 1),

--- a/pandas/tests/series/test_rank.py
+++ b/pandas/tests/series/test_rank.py
@@ -497,6 +497,7 @@ def test_rank_first_pct(dtype, ser, exp):
         assert_series_equal(result, expected)
 
 
+@pytest.mark.single
 def test_pct_max_many_rows():
         # GH 18271
         s = Series(np.arange(2**24 + 1))

--- a/pandas/tests/test_algos.py
+++ b/pandas/tests/test_algos.py
@@ -1462,6 +1462,7 @@ class TestRank(object):
         with pytest.raises(TypeError, match=msg):
             algos.rank(arr)
 
+    @pytest.mark.single
     @pytest.mark.parametrize('values', [
         np.arange(2**24 + 1),
         np.arange(2**25 + 2).reshape(2**24 + 1, 2)],

